Rudder-Kochava/Classes/RudderKochavaIntegration.m

```diff
@@ -78,215 +64,163 @@ - (void)flush {
+        [eventProperties removeObjectForKey:KeyCurrency];
     }
     return eventProperties;
 }
 
 
--(NSMutableDictionary*) setProductProperties: (NSMutableDictionary*) eventProperties withEvent: (KVAEvent*) event{
-    if (eventProperties[@"name"])
-    {
+- (NSMutableDictionary*) setProductProperties: (NSMutableDictionary*) eventProperties withEvent: (KVAEvent*) event{
+    if (eventProperties[@"name"]) {
         event.nameString = (NSString*)eventProperties[@"name"];
         [eventProperties removeObjectForKey:@"name"];
     }
-    if (eventProperties[@"product_id"])
-    {
-        event.contentIdString = (NSString*)eventProperties[@"product_id"];
-        [eventProperties removeObjectForKey:@"product_id"];
-        return eventProperties;
-    }
-    if (eventProperties[@"productId"])
-    {
-        event.contentIdString = (NSString*)eventProperties[@"productId"];
-        [eventProperties removeObjectForKey:@"productId"];
+    if (eventProperties[KeyProductId]) {
+        event.contentIdString = (NSString*)eventProperties[KeyProductId];
+        [eventProperties removeObjectForKey:KeyProductId];
     }
     return eventProperties;
 }
 
--(void) setProductsProperties: (NSMutableDictionary*) eventProperties withEvent: (KVAEvent*) event {
-    if (eventProperties[@"products"])
-    {
-        NSArray *products = eventProperties[@"products"];
+- (void) setProductsProperties: (NSMutableDictionary*) eventProperties withEvent: (KVAEvent*) event {
+    if (eventProperties[KeyProducts]) {
+        NSArray *products = eventProperties[KeyProducts];
         NSString* productNames = [self getProductProperties:products type:@"name"];
-        if (productNames)
-        {
+        if (productNames) {
             event.nameString = productNames;
         }
-        NSString* productIds = [self getProductProperties:products type:@"product_id"];
-        if (productIds)
-        {
+        NSString* productIds = [self getProductProperties:products type:KeyProductId];
+        if (productIds) {
             event.contentIdString = productIds;
         }
-        [eventProperties removeObjectForKey:@"products"];
+        [eventProperties removeObjectForKey:KeyProducts];
     }
 }
 
-- (NSString*) getProductProperties: (NSArray*) products type:(NSString*) type
-{
+- (NSString*) getProductProperties: (NSArray*) products type:(NSString*) type {
     NSMutableArray<NSString*>* productProperties = [[NSMutableArray alloc] init];
-    if (products)
-    {
-        for(NSDictionary *product in products)
-        {
-            if (product[type])
-            {
+    if (products) {
+        for(NSDictionary *product in products) {
+            if (product[type]) {
                 [productProperties addObject:product[type]];
             }
-            if ([type isEqual:@"product_id"] && product[@"productId"])
-            {
-                [productProperties addObject:product[@"productId"]];
-            }
```

Comment:
Let's have same behaviour on both android and ios

Reply:
I've refactored this logic.